pkgs/racket-doc/syntax/scribblings/parse/experimental.scrbl

```diff
@@ -346,4 +346,37 @@ the context above; instead, @racket[let-values] would report an
 invalid binding list.
 }
 
+@deftogether[
+ ((defthing prop:template-metafunction struct-type-property?)
+  (defthing template-metafunction? (-> any/c boolean?))
+  (defthing template-metafunction-accessor (-> template-metafunction? identifier?)))]{
```

Comment:
I think it might be better to only export `prop:template-metafunction`, since that’s what most of the other similar properties do, and it hides details of the implementation. Also, it doesn’t look like these are actually exported from `racket/base` anywhere.

Reply:
Can one extract the template-metafunction from an identifier with just the `prop:template-metafunction` or should I export a higher-level `identifier? → procedure?` extractor? (in order to manually expand the metafunction, I think this can be useful if you want to interleave metafunction expansion with some custom rewriting of the stx object.)

I haven't written struct-type-properties for a while, so I don't remember :).

Reply:
You can’t do that, but I think it would be better to provide a separate `expand-template` function if you really wanted, which would handle things like `~?` and `~@` properly. I don’t think that’s necessary, though, since you can already manually expand a metafunction by `local-expand`ing a use of `syntax`, so it seems unnecessarily complicated (and I think expanding a metafunction by hand seems potentially indicative of something maybe being a bad idea…).

Reply:
They shouldn't be exported from `racket/base` but from `syntax/parse/experimental/template`. However, currently they're not exported from there either, and they're called `prop:metafunction` and `metafunction?` when they should be called `prop:template-metafunction` and `template-metafunction?`.

I started a branch to fix that
https://github.com/racket/racket/compare/master...AlexKnauth:prop-template-metafunction-2

Reply:
@AlexKnauth's branch has been merged into this PR a while ago, just double-checked and it seems that all is up-to-date so I'm marking this discussion as resolved --- please do reopen if I was mistaken :)